### PR TITLE
Dashboard Sprint - 10

### PR DESCRIPTION
**Pull request by dbulatovicx32**

### CODE REVIEW

File: src/modules/documentExecution/dashboard/DashboardHelpers.ts [hunk 1]
```diff
@@ -142,4 +142,19 @@ export const getFormattedOutputParameters = (documentOutputParameters: IDashboar
             type: documentOutputParameter.type
         }
     })
+}
+
+export const apllyDashboardViewToModel = (dashboardModel: IDashboard, view: IDashboardView | null) => {
```
Comment (n3ils) [inline src/modules/documentExecution/dashboard/DashboardHelpers.ts:147]: Could you please fix the typo in the name? Thank you

File: src/modules/documentExecution/dashboard/DataProxyHelper.ts [hunk 2]
```diff
@@ -578,8 +618,10 @@ const formatChartWidgetForGet = (propWidget: IWidget, dataset: IDashboardDataset
             })
         } else {
             //FIRST CATEGORY LOGIC - TODO: Make it grab the drilldown Category instead of the first one.
+            // TODO - Darko Drill down level
+            // uzimam kategoriju na osnovu drill nivoa, ako ga nema (pocetno stanje) uzimam ovako kako smo radili, ali bi to TREBALO da je uvek prva kolona
```
Comment (n3ils) [inline src/modules/documentExecution/dashboard/DataProxyHelper.ts:622]: Reminder: remove comments when no longer useful

File: src/modules/documentExecution/dashboard/DashboardRenderer.vue [hunk 3]
```diff
@@ -65,14 +66,30 @@ export default defineComponent({
...
+            return `<style>${this.dashboardModel?.configuration?.cssToRender}</style>`
+        }
     },
     mounted() {
         this.dashboardModel = this.model ?? {}
         if (!this.dashboardModel.sheets) this.dashboardModel.sheets = []
         if (this.dashboardModel.sheets.length === 0) this.dashboardModel.sheets.push({ label: 'new sheet', widgets: { lg: [] } })
-    },
 
+        console.log(this.dashboardModel.configuration.background)
```
Comment (n3ils) [inline src/modules/documentExecution/dashboard/DashboardRenderer.vue:91]: Could you please remove the console.log? Thank you.

File: src/modules/documentExecution/main/DocumentExecution.vue [hunk 4]
```diff
@@ -182,6 +184,8 @@ import DashboardController from '../dashboard/DashboardController.vue'
 import Dialog from 'primevue/dialog'
 import descriptor from './DocumentExecutionDescriptor.json'
 import UserFunctionalitiesConstants from '@/UserFunctionalitiesConstants.json'
+import WorkspaceFolderPickerDialog from './dialogs/workspaceFolderPickerDialog/WorkspaceFolderPickerDialog.vue'
+import mockedView from './mockedView.json'
```
Comment (n3ils) [inline src/modules/documentExecution/main/DocumentExecution.vue:188]: Can you remove it if unused, please? Thank you.

**Review by n3ils — CHANGES_REQUESTED** [review]

File: src/modules/documentExecution/dashboard/DashboardHelpers.ts (hunk 1 above)
Comment (BojanSovticEngIT) [reply src/modules/documentExecution/dashboard/DashboardHelpers.ts]: Fixed

File: src/modules/documentExecution/main/DocumentExecution.vue (hunk 4 above)
Comment (BojanSovticEngIT) [reply src/modules/documentExecution/main/DocumentExecution.vue]: Fixed

File: src/modules/documentExecution/dashboard/widget/WidgetController.vue [hunk 5]
```diff
@@ -154,12 +160,21 @@ export default defineComponent({
...
             emitter.off('setWidgetLoading', this.setWidgetLoading)
         },
-        onWidgetRightClick(event) {
-            const contextMenu = this.$refs.contextMenu as any
-            contextMenu?.show(event)
+        onWidgetRightClick() {
+            // TODO: I dont think we need this anymore, remove?
+            // const contextMenu = this.$refs.contextMenu as any
+            // contextMenu?.show(event)
         },
```
[the remark this answers is not in the document]
Comment (dbulatovicx32) [reply src/modules/documentExecution/dashboard/widget/WidgetController.vue]: Removed

File: src/modules/documentExecution/dashboard/widget/TableWidget/PaginatorRenderer.vue [hunk 6]
```diff
@@ -20,15 +22,31 @@ export default defineComponent({
...
+                return {
+                    '--color': this.propWidget.settings.style.paginator.color,
+                    '--background-color': this.propWidget.settings.style.paginator['background-color'],
+                    '--justify-content': this.propWidget.settings.style.paginator['justify-content']
+                }
+            else
+                return {
+                    '--color': 'gray',
+                    '--background-color': 'white',
+                    '--justify-content': 'flex-end'
```
[the remark this answers is not in the document]
Comment (dbulatovicx32) [reply src/modules/documentExecution/dashboard/widget/TableWidget/PaginatorRenderer.vue:46]: Yes they should be scoped for specific widget.

File: src/modules/documentExecution/dashboard/generalSettings/background/DashboardBackground.vue [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+        return {
+            descriptor,
+            background: {} as IBackground
+        }
+    },
+    watch: {},
+    created() {
+        this.background = this.dashboardModelProp.configuration.background as IBackground
+        console.log('dashboardModel', this.dashboardModelProp)
+        console.log('background', this.background)
```
[the remark this answers is not in the document]
Comment (dbulatovicx32) [reply src/modules/documentExecution/dashboard/generalSettings/background/DashboardBackground.vue]: Removed

File: src/modules/documentExecution/dashboard/dataset/DatasetEditorDataTab/DatasetEditorDataTab.vue [hunk 8]
```diff
@@ -50,6 +56,27 @@ export default defineComponent({
...
+        },
+        async loadDataset(datasetLabel: string) {
+            await this.$http
+                .get(import.meta.env.VITE_RESTFUL_SERVICES_PATH + `1.0/datasets/${datasetLabel}`)
+                .then((response: AxiosResponse<any>) => {
+                    this.datasetToPreview = response.data[0]
+                })
+                .catch(() => {})
+        },
+        async previewSelectedDataset() {
```
[the remark this answers is not in the document]
Comment (dbulatovicx32) [reply src/modules/documentExecution/dashboard/dataset/DatasetEditorDataTab/DatasetEditorDataTab.vue:68]: Removed

File: src/modules/documentExecution/dashboard/DashboardRenderer.vue (hunk 3 above)
Comment (dbulatovicx32) [reply src/modules/documentExecution/dashboard/DashboardRenderer.vue]: Removed

File: src/modules/documentExecution/dashboard/DataProxyHelper.ts (hunk 2 above)
Comment (dbulatovicx32) [reply src/modules/documentExecution/dashboard/DataProxyHelper.ts]: Removed

**Review by n3ils — APPROVED** [review]

**Pull request merged**